core/lib/spree/service_module.rb

```diff
@@ -0,0 +1,64 @@
+module Spree
+  module ServiceModule
+    class MethodNotImplemented < StandardError; end
+    class WrongDataPassed < StandardError; end
+    class NonCallablePassedToRun < StandardError; end
+    class CallMethodNotImplemented < StandardError; end
+
+    Result = Struct.new(:success, :value) do
+      def success?
+        success
+      end
+
+      def failure?
+        !success
+      end
+    end
+
+    module Base
+      def call(input = nil)
+        input ||= {}
+        @_passed_input = Result.new(true, input)
```

Comment:
What's the point of using `_`? I know this is used in other languages to indicate private / internal stuff but it's not common in Ruby AFAIK

Reply:
I don't want to collide if someone names instance variable like that, but I just realized, that people shouldn't use instance variables at all